chezbetty/datalayer.py

```diff
@@ -179,7 +179,15 @@ def purchase(user, account, items):
     # TODO: Parameterize
     discount = Decimal(0)
     if user.balance > 20.0:
-        discount = Decimal('0.05')
+        #give discounts based on user type; TODO: exact discounts should be adjusted as involvement changes
+        if user.role is "volunteer":
```

Comment:
This is not a safe way of performing string equality in python. `is` checks identify equality rather than value equality. There's some background on this here: https://dbader.org/blog/difference-between-is-and-equals-in-python. It's also pretty ugly to have this as a fraction for some and decimal in other places. Should probably just be `0.05` everywhere.

Reply:
I will update to use "==" instead of "is".  The fraction is because we want to give the other roles wholesale prices (which is not the same as 20% off the sale price), but we want users in good standing to have 5% off the sale price (which is not the same as a 15% markup instead of a 20% markup).  Do you have an idea on how to achieve this without having a fraction?  It seemed to be the minimal number of code changes required to achieve the desired result.